### PR TITLE
Adds TimeStamp UI Rule Updates

### PR DESCRIPTION
**Pull request by zeph-cohen10**
This PR adjusts the timestamp rendering to be a bit more dynamic / flexible based on the current designs and video durations.

### CODE REVIEW

File: Source/Configuration/YPImagePickerConfiguration.swift [hunk 1]
```diff
@@ -309,7 +309,7 @@ public struct YPConfigVideo {
     public var trimmerMinDuration: Double = 3.0
 
     /// Defines if the trimmer will display the timestamp user interface.
-    public var displayTrimmerWithTimeStamps: Bool = false
+     public var displayTrimmerWithTimeStamps: Bool = false
```
Comment (will-ltk) [inline Source/Configuration/YPImagePickerConfiguration.swift:312]: nit: extra space

File: Source/Filters/Trim/YPTimeStampScrollView.swift [hunk 2]
```diff
@@ -31,41 +48,83 @@ class YPTimeStampScrollableView: UIScrollView {
...
-            // On every 6th view (not counting the starting view) we want to render the video timestamp at that point.
-            if i % 6 == 0, let asset = asset {
-                contentViewSubView.timeStampText = getTime(from: CGFloat(xPosition), for: asset)?.durationText
-            }
-
-            contentViewSubView.frame = CGRect(x: (i * width) + startOffset, y: 0, width: width, height: 23)
+            // We want the first 0:00 dot to line up just after the left handle bar. This offset is to account for the handle bar width and
+            // start the first timestamp just to the right of the left handle.
+            
+            //We also need to move the xPosition forward by the same offset value used to pull the time stamp UI back to the left. This pullback is
```
Comment (will-ltk) [inline Source/Filters/Trim/YPTimeStampScrollView.swift:121]: super nit: extra space needed before "We" for conformity.

File: Source/Filters/Trim/YPTimeStampScrollView.swift [hunk 3]
```diff
@@ -31,41 +48,83 @@ class YPTimeStampScrollableView: UIScrollView {
...
+                timeBarColor: range.shouldRenderTimeStamp ? timeBarLargeCircleColor : timeBarSmallCircleColor
             )
 
-            let xPosition = i * width
-
-            if i == 0, let asset = asset {
-                contentViewSubView.shouldRenderBoldCircle = true
-                contentViewSubView.timeStampText = getTime(from: CGFloat(xPosition), for: asset)?.durationText
-            }
+            guard var xPosition = getPosition(from: range.startTime) else { continue }
```
Comment (will-ltk) [inline Source/Filters/Trim/YPTimeStampScrollView.swift:112]: Should we consider logging if the guard returns here? Wondering if returning causes the UI to break.

File: Source/Configuration/YPImagePickerConfiguration.swift (hunk 1 above)
Comment (zeph-cohen10) [reply Source/Configuration/YPImagePickerConfiguration.swift]: @will-ltk Updated

File: Source/Filters/Trim/YPTimeStampScrollView.swift (hunk 2 above)
Comment (zeph-cohen10) [reply Source/Filters/Trim/YPTimeStampScrollView.swift]: @will-ltk Updated this too.

**Review by will-ltk — APPROVED** [review]

**Pull request merged**